src/pointtopoint.jl

```diff
@@ -447,6 +447,7 @@ $(_doc_external("MPI_Waitall"))
 """
 function Waitall!(reqs::Vector{Request})
     count = length(reqs)
+    count == 0 && return []
```

Comment:
Make it type-stable.
```suggestion
    count == 0 && return Status[]
```